zebra-chain/src/orchard/tree.rs

```diff
@@ -317,6 +339,46 @@ impl NoteCommitmentTree {
         }
     }
 
+    /// Returns true if the most recently appended leaf completes the subtree
+    pub fn is_complete_subtree(&self) -> bool {
+        self.inner.value().map_or(false, |x| {
+            x.position()
+                .is_complete_subtree(TRACKED_SUBTREE_HEIGHT.into())
+        })
+    }
+
+    /// Returns subtree index at [`TRACKED_SUBTREE_HEIGHT`]
+    pub fn subtree_index(&self) -> u16 {
+        (self.position() >> TRACKED_SUBTREE_HEIGHT)
+            .try_into()
+            .expect("always within u16")
+    }
+
+    /// Returns subtree root, if any
+    pub fn subtree_root(&self) -> Option<Node> {
+        self.is_complete_subtree().then_some(())?;
+        Some(
+            self.inner
+                .value()?
+                .root(Some(TRACKED_SUBTREE_HEIGHT.into())),
+        )
+    }
+
+    /// Returns the index of note completing the next subtree
+    pub fn next_complete_subtree_note_index(&self) -> usize {
+        // # Correctness
+        //
+        // 2^16 - (position % 2^16) should always be between 1..=2^16
+        (TRACKED_SUBTREE_SIZE - self.subtree_note_count() - 1)
+            .try_into()
+            .expect("should fit in usize")
+    }
```

Comment:
> What this method actually seems to do is return the number of remaining notes before the current subtree is complete

I had a lot of trouble naming this because it's the remaining notes before the current subtree is complete except when the last note to be appended completed a subtree, in which case it's 2^16 - 1.

I'll add a method that returns the Address so that the methods it provides can be used directly.